### DISCUSSION TITLE
If creating a(n encrypted) multi volume file, can I know which file will end up in which volume?

### DISCUSSION THREAD

**Discussion by OWL4C** [miurahr/py7zr#578] [General]
If adding x files from a list to a multi volume archive, can I somehow create a dictionary that tracks which file ended up in which volume? So if i have to change one file, can i find out which volume i will have to recreate?

**Comment by miurahr** [comment]
From a nature of  7zip encryption format, you need to recreate all the volumes. There are several reasons, but one significant thing is that the secret (internal) key is created from your passphrase and random number generator in every processing time.  
The salt, random number,  is stored in header. When you regenerated encrypted 7zip file, even if you don't change files, most of data will be different.
